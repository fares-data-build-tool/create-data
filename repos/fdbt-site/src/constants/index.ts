export const OPERATOR_COOKIE: string = 'fdbt-operator-cookie';

export const SERVICE_COOKIE: string = 'fdbt-service-cookie';

export const FARETYPE_COOKIE: string = 'fdbt-faretype-cookie';

export const FEEDBACK_LINK: string = 'mailto:fdbt@transportforthenorth.com';

export const GOVUK_LINK: string = 'https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/';

export const CSV_INPUT_METHOD_COOKIE: string = 'fdbt-csv-input-method-cookie';

export const MANUAL_INPUT_METHOD_COOKIE: string = 'fdbt-manual-input-method-cookie';

export const INT_MAP_INPUT_METHOD_COOKIE: string = 'fdbt-interactive-map-input-method-cookie';
