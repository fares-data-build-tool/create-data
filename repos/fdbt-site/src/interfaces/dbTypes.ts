import { CompanionInfo, DbTimeBand, TimeRestrictionDay } from './matchingJsonTypes';

export interface FullGroupPassengerType {
    id: number;
    name: string;
    groupPassengerType: GroupPassengerType;
}

export interface SinglePassengerType {
    id: number;
    name: string;
    passengerType: PassengerType;
}

export interface PassengerType {
    passengerType: string;
    ageRange?: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proof?: string;
    proofDocuments?: string[];
    id: number;
}

export interface MyFaresProduct {
    id: number;
    lineId: string;
    matchingJsonLink: string;
    startDate: string;
    endDate?: string;
    servicesRequiringAttention?: string[];
    fareTriangleModified?: string;
}

export type RawMyFaresProduct = Omit<MyFaresProduct, 'servicesRequiringAttention'> & {
    servicesRequiringAttention: string;
};

export interface DbProduct {
    id: string;
    matchingJsonLink: string;
    lineId: string;
    startDate: string;
    endDate?: string;
}

export type MyFaresOtherProduct = Omit<MyFaresProduct, 'lineId'>;

export interface ServiceQueryData {
    operatorShortName: string;
    serviceDescription: string;
    lineName: string;
    startDate: string;
    lineId: string;
    fromAtcoCode: string;
    toAtcoCode: string;
    fromCommonName: string;
    toCommonName: string;
    journeyPatternId: string;
    order: string;
    direction: string;
    fromSequenceNumber: number;
    toSequenceNumber: number;
    inboundDirectionDescription: string;
    outboundDirectionDescription: string;
}

export type StopPoint = {
    stopPointRef: string;
    commonName: string;
    sequenceNumber?: number;
};

export interface RawJourneyPattern {
    orderedStopPoints: StopPoint[];
    direction: string;
}

export interface RawService {
    serviceDescription: string;
    operatorShortName: string;
    journeyPatterns: RawJourneyPattern[];
    lineId: string;
    lineName: string;
    startDate: string;
    endDate?: string;
    inboundDirectionDescription: string;
    outboundDirectionDescription: string;
}

export interface GroupPassengerType {
    name: string;
    maxGroupSize: string;
    companions: CompanionInfo[];
}

export interface RawSalesOfferPackage {
    name: string;
    description: string;
    id: number;
    purchaseLocations: string;
    paymentMethods: string;
    ticketFormats: string;
}
export interface GroupPassengerTypeDb {
    id: number;
    name: string;
    groupPassengerType: GroupPassengerTypeReference;
}

export interface CompanionReference {
    id: number;
    name: string;
    minNumber?: string;
    maxNumber: string;
}

export interface GroupPassengerTypeReference {
    name: string;
    maxGroupSize: string;
    companions: CompanionReference[];
}

export interface DbTimeRestriction {
    day: TimeRestrictionDay;
    timeBands: DbTimeBand[];
}

export interface ServiceDetails {
    fromAtcoCode: string;
    toAtcoCode: string;
    direction: string;
    serviceId: string;
}
