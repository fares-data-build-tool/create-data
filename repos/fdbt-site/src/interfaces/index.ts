export interface PeriodProductType {
    uuid: string;
    productName: string;
    productPrice: string;
    productNameError: string;
    productPriceError: string;
}
