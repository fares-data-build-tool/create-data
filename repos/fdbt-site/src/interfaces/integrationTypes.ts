export type ExportLambdaBody = { paths: string[]; noc: string; exportPrefix?: string };

export type ZipperLambdaBody = { exportName: string; noc: string };
