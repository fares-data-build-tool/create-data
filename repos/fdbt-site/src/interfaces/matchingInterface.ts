import { Stop } from '../data/auroradb';
import { PassengerDetails, ServicesInfo, BasicService, Product } from '.';
import { UserFareStages } from '../data/s3';

export interface Price {
    price: string;
    fareZones: string[];
}

export interface MatchingInfo {
    service: BasicService;
    userFareStages: UserFareStages;
    matchingFareZones: MatchingFareZones;
}

export interface InboundMatchingInfo {
    inboundUserFareStages: UserFareStages;
    inboundMatchingFareZones: MatchingFareZones;
}

export interface MatchingWithErrors {
    error: boolean;
    selectedFareStages: string[];
}

export interface MatchingFareZonesData {
    name: string;
    stops: Stop[];
    prices: Price[];
}

export interface MatchingFareZones {
    [key: string]: MatchingFareZonesData;
}

export interface MatchingBaseData {
    type: string;
    lineName: string;
    nocCode: string;
    operatorShortName: string;
    serviceDescription: string;
    email: string;
    uuid: string;
}

export interface MatchingData extends MatchingBaseData {
    fareZones: {
        name: string;
        stops: Stop[];
        prices: {
            price: string;
            fareZones: string[];
        }[];
    }[];
}

export interface MatchingReturnData extends MatchingBaseData {
    outboundFareZones: {
        name: string;
        stops: Stop[];
        prices: {
            price: string;
            fareZones: string[];
        }[];
    }[];
    inboundFareZones: {
        name: string;
        stops: Stop[];
        prices: {
            price: string;
            fareZones: string[];
        }[];
    }[];
}

export interface MatchingPeriodData extends PassengerDetails {
    operatorName: string;
    type: string;
    nocCode: string;
    products: Product[];
    selectedServices?: ServicesInfo[];
    zoneName?: string;
    stops?: Stop[];
    email: string;
    uuid: string;
}
