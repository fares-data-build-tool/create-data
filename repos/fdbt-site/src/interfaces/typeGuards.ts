import { PeriodTicket, PointToPointTicket, PeriodMultipleServicesTicket, FlatFareTicket, PeriodGeoZoneTicket } from '.';

export const isNotEmpty = <T>(value: T | null | undefined): value is T => value !== null && value !== undefined;

export const isPeriodTicket = (ticket: PeriodTicket | PointToPointTicket): ticket is PeriodTicket =>
    (ticket as PeriodTicket).products?.[0]?.productName !== undefined;

export const isMultipleServicesTicket = (
    ticket: PeriodTicket | PointToPointTicket,
): ticket is PeriodMultipleServicesTicket | FlatFareTicket =>
    (ticket as PeriodMultipleServicesTicket).selectedServices !== undefined;

export const isPointToPointTicket = (ticket: PeriodTicket | PointToPointTicket): ticket is PointToPointTicket =>
    (ticket as PointToPointTicket).lineName !== undefined;

export const isGeoZoneTicket = (ticket: PeriodTicket | PointToPointTicket): ticket is PeriodGeoZoneTicket =>
    (ticket as PeriodGeoZoneTicket).zoneName !== undefined;
