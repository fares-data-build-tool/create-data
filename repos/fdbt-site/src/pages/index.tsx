import '../design/Pages.scss';
import React, { ReactElement } from 'react';
import { NextPage } from 'next';
import Layout from '../layout/Layout';

const title = 'Fares data build tool';
const description = 'Fares data build tool is a service that allows you to generate data in NeTEx format';

const Home: NextPage = (): ReactElement => (
    <Layout title={title} description={description}>
        <main className="govuk-main-wrapper app-main-class" id="main-content" role="main">
            <h1 className="govuk-heading-xl">Fares data build tool</h1>
            <p className="govuk-body-l">Use this service to:</p>
            <div className="govuk-body-l">
                <ul className="index-page-list">
                    <li>Add fares to a newly registered bus service</li>
                    <li>Update fares on an existing bus service</li>
                    <li>Generate fares data for services in NeTEx format</li>
                </ul>
            </div>

            <a
                href="/operator"
                role="button"
                draggable="false"
                className="govuk-button govuk-button--start"
                data-module="govuk-button"
                id="start-now-button"
            >
                Start now
                <svg
                    className="govuk-button__start-icon"
                    xmlns="http://www.w3.org/2000/svg"
                    width="17.5"
                    height="19"
                    viewBox="0 0 33 40"
                    role="presentation"
                    focusable="false"
                >
                    <path fill="currentColor" d="M0 0h13l20 20-20 20H0l20-20z" />
                </svg>
            </a>
        </main>
    </Layout>
);

Home.getInitialProps = (): {} => {
    return {};
};

export default Home;
