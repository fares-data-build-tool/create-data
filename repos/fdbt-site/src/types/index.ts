export type ErrorInfo = {
    errorMessage: string;
};

export type Uuid = {
    uuid: string;
};
