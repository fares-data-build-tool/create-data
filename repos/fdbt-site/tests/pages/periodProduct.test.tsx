/* eslint-disable global-require */

import * as React from 'react';
import { shallow } from 'enzyme';
import PeriodProduct from '../../src/pages/periodProduct';
import { PeriodProductType } from '../../src/interfaces';

const mockPeriodProduct: PeriodProductType = {
    productPrice: '',
    productName: '',
    productNameError: false,
    productPriceError: false,
    uuid: '',
};

describe('pages', () => {
    describe('periodProduct', () => {
        it('should render correctly', () => {
            const tree = shallow(<PeriodProduct product={mockPeriodProduct} operator="bus company" />);
            expect(tree).toMatchSnapshot();
        });
    });
});
