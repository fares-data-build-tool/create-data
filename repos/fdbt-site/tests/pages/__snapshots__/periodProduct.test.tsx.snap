// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages periodProduct should render correctly 1`] = `
<Layout
  description="Fare Type selection page of the Fares data build tool"
  title="FareType - Fares data build tool"
>
  <main
    className="govuk-main-wrapper app-main-class"
    id="main-content"
    role="main"
  >
    <form
      action="/api/periodProduct"
      method="post"
    >
      <div
        className="govuk-form-group"
      >
        <fieldset
          aria-describedby="period-product-page-heading"
          className="govuk-fieldset"
        >
          <legend
            className="govuk-fieldset__legend govuk-fieldset__legend--xl"
          >
            <h1
              className="govuk-fieldset__heading"
              id="period-product-page-heading"
            >
              Enter your product details
            </h1>
          </legend>
          <span
            className="govuk-hint"
            id="service-operator-hint"
          >
            bus company
             - 
            Test Zone
          </span>
        </fieldset>
        <div
          className="govuk-form-group "
        >
          <label
            className="govuk-label"
            htmlFor="periodProductName"
          >
            Product Name
          </label>
          <span
            className="govuk-hint"
            id="product-name-hint"
          >
            Please enter the name of your product
          </span>
          <span
            className="govuk-error-message"
            id="product-price-error"
          >
            <span
              className="govuk-visually-hidden"
            />
          </span>
          <input
            className="govuk-input govuk-input--width-30 govuk-product-name-input__inner__input  "
            defaultValue=""
            id="periodProductName"
            maxLength={50}
            name="periodProductNameInput"
            type="text"
          />
        </div>
        <div
          className="govuk-form-group "
        >
          <label
            className="govuk-label"
            htmlFor="periodProductPrice"
          >
            Product Price
          </label>
          <span
            className="govuk-hint"
            id="product-price-hint"
          >
            For example, £2.99
          </span>
          <span
            className="govuk-error-message"
            id="product-price-error"
          >
            <span
              className="govuk-visually-hidden"
            />
          </span>
          <div
            className="govuk-currency-input"
          >
            <div
              className="govuk-currency-input__inner"
            >
              <span
                className="govuk-currency-input__inner__unit"
              >
                £
              </span>
              <input
                aria-label="Enter amount in pounds"
                className="govuk-input govuk-input--width-10 govuk-currency-input__inner__input "
                data-non-numeric={true}
                defaultValue=""
                id="periodProductPrice"
                name="periodProductPriceInput"
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <input
        className="govuk-button govuk-button--start"
        id="continue-button"
        type="submit"
        value="Continue"
      />
    </form>
  </main>
</Layout>
`;
